In [36]:
import pandas as pd


In [37]:
# читаю датафрейм только с нужными столбцами
df = pd.read_excel('/content/Тестовое.xlsx', sheet_name = 2, usecols=[0, 1, 3, 4, 7, 8])
df.head()


ДатаПродажи  НомерЧека       Штрихкод   ТоварнаяГруппа  ОбщаяСумма  \
0  2023-04-01          1  1078601904686     СИ КОЛЬЦО БК     1350.90   
1  2023-04-01          1  1078601904686        БК КОЛЬЦА     5449.08   
2  2023-04-01          1  1078601904686        ИФ КОЛЬЦА     5612.40   
3  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10510.50   
4  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10040.70   

   Себестоимость  
0       751.9850  
1      3408.8760  
2      4255.4025  
3      6233.2185  
4      6281.3115

In [38]:
# проверяю типы данных
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 139997 entries, 0 to 139996
Data columns (total 6 columns):
 #   Column          Non-Null Count   Dtype         
---  ------          --------------   -----         
 0   ДатаПродажи     139997 non-null  datetime64[ns]
 1   НомерЧека       139997 non-null  int64         
 2   Штрихкод        139997 non-null  int64         
 3   ТоварнаяГруппа  139997 non-null  object        
 4   ОбщаяСумма      139997 non-null  float64       
 5   Себестоимость   139997 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(2), object(1)
memory usage: 6.4+ MB


In [39]:
# обрезаю датафрейм по дате и сумам продаж
df = df.drop(df[df['ДатаПродажи'] < '2023-04-01'].index)
df = df.drop(df[df['ДатаПродажи'] > '2023-04-10'].index)
df = df.drop(df[df['ОбщаяСумма'] <= 0].index)
df = df.drop(df[df['Себестоимость'] <= 0].index)
df.head()

ДатаПродажи  НомерЧека       Штрихкод   ТоварнаяГруппа  ОбщаяСумма  \
0  2023-04-01          1  1078601904686     СИ КОЛЬЦО БК     1350.90   
1  2023-04-01          1  1078601904686        БК КОЛЬЦА     5449.08   
2  2023-04-01          1  1078601904686        ИФ КОЛЬЦА     5612.40   
3  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10510.50   
4  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10040.70   

   Себестоимость  
0       751.9850  
1      3408.8760  
2      4255.4025  
3      6233.2185  
4      6281.3115

In [40]:
#  проверяю изменения
df.describe()

ДатаПродажи     НомерЧека      Штрихкод  \
count                          84614  84614.000000  8.461400e+04   
mean   2023-04-05 12:31:44.330949376   3287.442870  1.078602e+12   
min              2023-04-01 00:00:00      1.000000  1.078602e+12   
25%       2023-04-02 19:35:43.500000   1519.000000  1.078602e+12   
50%              2023-04-05 13:46:53   3514.000000  1.078602e+12   
75%              2023-04-07 18:20:58   4491.000000  1.078602e+12   
max              2023-04-10 00:00:00  10199.000000  1.078602e+12   
std                              NaN   1991.852385  1.605234e+00   

          ОбщаяСумма  Себестоимость  
count   84614.000000   84614.000000  
mean     9152.877193    3722.571357  
min         1.200000      12.420000  
25%      1488.000000     521.950500  
50%      5280.000000    2016.663000  
75%     13080.000000    5165.104250  
max    515940.000000  296546.624500  
std     12021.462844    5798.469898

In [41]:
# создаю колонку с значениями наценки
df['Наценка'] = df.apply(lambda x: x['ОбщаяСумма'] - x['Себестоимость'], axis=1)
df.head()

ДатаПродажи  НомерЧека       Штрихкод   ТоварнаяГруппа  ОбщаяСумма  \
0  2023-04-01          1  1078601904686     СИ КОЛЬЦО БК     1350.90   
1  2023-04-01          1  1078601904686        БК КОЛЬЦА     5449.08   
2  2023-04-01          1  1078601904686        ИФ КОЛЬЦА     5612.40   
3  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10510.50   
4  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10040.70   

   Себестоимость    Наценка  
0       751.9850   598.9150  
1      3408.8760  2040.2040  
2      4255.4025  1356.9975  
3      6233.2185  4277.2815  
4      6281.3115  3759.3885

In [42]:
# создаю категориальную колонку 1 - золото, 0 - серебро
df['Металл'] = df['ТоварнаяГруппа'].apply(lambda x: 0 if 'СИ' in x
                                          else 1)
df.head()

ДатаПродажи  НомерЧека       Штрихкод   ТоварнаяГруппа  ОбщаяСумма  \
0  2023-04-01          1  1078601904686     СИ КОЛЬЦО БК     1350.90   
1  2023-04-01          1  1078601904686        БК КОЛЬЦА     5449.08   
2  2023-04-01          1  1078601904686        ИФ КОЛЬЦА     5612.40   
3  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10510.50   
4  2023-04-01          1  1078601904686  БК КОЛЬЦА ОБРУЧ    10040.70   

   Себестоимость    Наценка  Металл  
0       751.9850   598.9150       0  
1      3408.8760  2040.2040       1  
2      4255.4025  1356.9975       1  
3      6233.2185  4277.2815       1  
4      6281.3115  3759.3885       1

In [43]:
# группирую по категории и вывожу средние значения наценки
df_group = df.groupby(['Металл']).agg({'Наценка': 'mean'})
df_group

Наценка
Металл             
0       1293.690315
1       8634.467263

Средняя наценка для серебра: 1294р
Средняя наценка для золота: 8634р